src/smartactions/SchedulingBase.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+import { IERC20 } from "forge-std/interfaces/IERC20.sol";
+import { IERC7579Execution } from "modulekit/Accounts.sol";
+import { ERC7579ExecutorBase } from "modulekit/Modules.sol";
+
+abstract contract SchedulingBase is ERC7579ExecutorBase {
```

Comment:
it seems this is only an executor. how would the automation be invoked with the user paying for the gas